crates/context_aware_config/src/api/context/handlers.rs

```diff
@@ -392,26 +407,35 @@ async fn update_override_handler(
     req: Json<PutReq>,
     mut db_conn: DbConnection,
     user: User,
+    #[cfg(feature = "high-performance-mode")] tenant: Tenant,
     tenant_config: TenantConfig,
 ) -> superposition::Result<HttpResponse> {
     let tags = parse_config_tags(custom_headers.config_tags)?;
-    db_conn.transaction::<_, superposition::AppError, _>(|transaction_conn| {
+    let mut version_id = 0;
+    let resp = db_conn.transaction::<_, superposition::AppError, _>(|transaction_conn| {
         let override_resp =
             override_helper(req, transaction_conn, true, &user, &tenant_config).map_err(
                 |err: superposition::AppError| {
                     log::info!("context put failed with error: {:?}", err);
                     err
                 },
             )?;
-        let version_id = add_config_version(&state, tags, transaction_conn)?;
+        version_id = add_config_version(&state, tags, transaction_conn)?;
         let mut http_resp = HttpResponse::Ok();
 
         http_resp.insert_header((
             AppHeader::XConfigVersion.to_string(),
             version_id.to_string(),
         ));
         Ok(http_resp.json(override_resp))
```

Comment:
The transaction closure can just return the version_id, HttpResponse construction can be moved out.